cmd/soroban-cli/src/commands/network/container/shared.rs

```diff
@@ -31,7 +31,18 @@ pub enum Error {
     UnsupportedURISchemeError { uri: String },
 }
 
-#[derive(ValueEnum, Debug, Clone, PartialEq)]
+#[derive(Debug, clap::Parser, Clone)]
+pub struct ContainerArgs {
+    /// Optional argument to specify the container name
+    #[arg(short = 'c', long, required_unless_present = "network")]
+    pub container_name: Option<String>,
+
```

Comment:
It seems everywhere you use this struct you also use `Network`, so it would make sense to add it here. Then you can add a `name` method to its `impl`

Reply:
That's a good call out. I didn't include `Network` in this struct because it is required for the `start` command, but is optional for `stop` and `logs`. Maybe that is confusing and could use a refactor/rethink. 🤔